### PR TITLE
Added Login part 

### PR DESCRIPTION
**Pull request by aritrasen12345**

### CODE REVIEW

File: src/components/app/landing/login/LoginModal.tsx [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+
+interface Props {
+  isOpen: boolean;
+  onClose: () => void;
+}
+
+const LoginModal = (props: Props) => {
+  const [formIsValid, setFormIsValid] = useState<boolean>(false);
+  const [enteredEmail, setEnteredEmail] = useState<string>('');
+  const [enteredPassword, setEnteredPassword] = useState<string>('');
```
Comment (Bijoy-007) [inline src/components/app/landing/login/LoginModal.tsx]: @aritrasen12345 When using antd Form component we do not need to maintain the state if we are not populating the form because on onFinish event we get the form details.

File: src/components/app/landing/login/LoginModal.tsx [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+
+  useEffect(() => {
+    setFormIsValid(
+      enteredEmail.includes('@') && enteredPassword.trim().length > 6,
+    );
+  }, [enteredEmail, enteredPassword]);
```
Comment (Bijoy-007) [inline src/components/app/landing/login/LoginModal.tsx]: @aritrasen12345 use antd form validation and control instead of manual. i.e. use `rules` props on                 `<Form.Item></Form.Item>`

File: src/components/app/landing/login/LoginModal.tsx [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+        >
+          <Form.Item
+            label="Email"
+            name="email"
+            rules={[{ required: true, message: 'Please input your email!' }]}
+          >
+            <Input
+              value={enteredEmail}
+              autoFocus={true}
+              onChange={(event) => setEnteredEmail(event.target.value)}
```
Comment (Bijoy-007) [inline src/components/app/landing/login/LoginModal.tsx]: @aritrasen12345 no need to maintain the manual state.

File: src/components/app/landing/login/LoginModal.tsx [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+            />
+          </Form.Item>
+          <Form.Item
+            label="Password"
+            name="password"
+            rules={[{ required: true, message: 'Please input your password!' }]}
+          >
+            <Input.Password
+              value={enteredPassword}
+              onChange={(event) => setEnteredPassword(event.target.value)}
```
Comment (Bijoy-007) [inline src/components/app/landing/login/LoginModal.tsx]: @aritrasen12345 refer above

File: src/components/app/landing/login/LoginModal.tsx [hunk 5]
```diff
@@ -0,0 +1,82 @@
...
+            />
+          </Form.Item>
+          <Button
+            shape="round"
+            type="primary"
+            htmlType="submit"
+            disabled={!formIsValid}
```
Comment (Bijoy-007) [inline src/components/app/landing/login/LoginModal.tsx]: @aritrasen12345 we do not need this if we have antd form control already

File: src/types/user/userDetails.ts [hunk 6]
```diff
@@ -0,0 +1,16 @@
...
+  email: string;
+  type: string;
+  joiningDate: Moment | null;
+  description: string;
+  phone: string;
+  isActive: boolean;
+  profileImage: string;
+  createdAt: Date;
+  label: string;
+}
```
Comment (Bijoy-007) [inline src/types/user/userDetails.ts]: @aritrasen12345 Is this interface relevant to this project? do we have `joiningDate` at all?

**Review by Bijoy-007 — CHANGES_REQUESTED** [review]

**Review by Prod-IP — COMMENTED** [review]
@aritrasen12345 use antd ` Form ` states instead of maintaining manual state

File: src/components/app/landing/login/LoginModal.tsx [hunk 7]
```diff
@@ -9,16 +9,12 @@ interface Props {
 }
 
 const LoginModal = (props: Props) => {
-  const [formIsValid, setFormIsValid] = useState<boolean>(false);
-  const [enteredEmail, setEnteredEmail] = useState<string>('');
-  const [enteredPassword, setEnteredPassword] = useState<string>('');
+  const [form] = Form.useForm();
+  const [, forceUpdate] = useState<object>({});
```
Comment (Prod-IP) [inline src/components/app/landing/login/LoginModal.tsx]: don't use state here 
make it a simple function 
`forceUpdate` use better var name

**Review by Prod-IP — COMMENTED** [review]
@aritrasen12345 check

File: src/components/app/landing/login/LoginModal.tsx (hunk 7 above)
Comment (Bijoy-007) [reply src/components/app/landing/login/LoginModal.tsx]: @aritrasen12345 What is the use of this state?

**Review by Prod-IP — COMMENTED** [review]
LGTM

**Review by Bijoy-007 — APPROVED** [review]
